### PR TITLE
Implement CreateBuild rpc

### PR DESCRIPTION
**Pull request by alexashley**
- converted `config` package tests to Ginkgo
- added commit id to `CreateBuildRequest` to populate revision id in the build occurrence

Example payload:

```json
{
	"commitId": "639550538727a298ed0757cdf8d1526d65eb5328",
	"repository": "https://github.com/rode/collector-build",
	"artifacts": [
		"ghcr.io/rode/collector-build@sha256:12345"
	]
}
```

Example response:
```json
{
  "buildOccurrenceId": "projects/rode/occurrences/fa9eb4fc-89d5-4d2f-99ce-2ca31eaf8099"
}
```

Resulting build occurrence:

```json
{
  "name": "projects/rode/occurrences/fa9eb4fc-89d5-4d2f-99ce-2ca31eaf8099",
  "resource": {
    "name": "",
    "uri": "git://github.com/rode/collector-build@639550538727a298ed0757cdf8d1526d65eb5328",
    "contentHash": null
  },
  "noteName": "projects/rode/notes/build_collector",
  "kind": "BUILD",
  "remediation": "",
  "createTime": "2021-03-23T20:52:36.837221100Z",
  "updateTime": null,
  "build": {
    "provenance": {
      "id": "167503e7-a9c5-4176-ae25-4c915373d105",
      "projectId": "projects/rode",
      "commands": [],
      "builtArtifacts": [
        {
          "checksum": "",
          "id": "ghcr.io/rode/collector-build@sha256:12345",
          "names": []
        }
      ],
      "createTime": "2021-03-23T20:52:36.823508Z",
      "startTime": null,
      "endTime": null,
      "creator": "",
      "logsUri": "",
      "sourceProvenance": {
        "artifactStorageSourceUri": "",
        "fileHashes": {},
        "context": {
          "git": {
            "url": "https://github.com/rode/collector-build",
            "revisionId": "639550538727a298ed0757cdf8d1526d65eb5328"
          },
          "labels": {}
        },
        "additionalContexts": []
      },
      "triggerId": "",
      "buildOptions": {},
      "builderVersion": ""
    },
    "provenanceBytes": ""
  }
}
```

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -35,5 +56,97 @@ func NewBuildCollectorServer(logger *zap.Logger, rode pb.RodeClient) *BuildColle
...
+	var artifacts []*provenance_go_proto.Artifact
+	for _, artifact := range request.Artifacts {
+		artifacts = append(artifacts, &provenance_go_proto.Artifact{
+			Id: artifact,
+		})
+	}
+
+	return &grafeas_go_proto.Occurrence{
+		Resource: &grafeas_go_proto.Resource{
+			Uri: fmt.Sprintf("git://%s%s@%s", repositoryURL.Host, repositoryURL.Path, request.CommitId),
```
Comment (aalsabag) [inline server/server.go:128]: @pickjasmine probably needs to be aware of this resource uri format

Comment (aalsabag) [reply server/server.go:128]: I don't think it is included in our current resource-utils

**Review by aalsabag — APPROVED** [review]
Lookin good. lgtm

File: server/server.go (hunk 1 above)
Comment (pickjasmine) [reply server/server.go:128]: I'll make an issue to handle this format in the UI `git://{host}{path}@{commit}`. thanks @aalsabag !

**Pull request merged**